splicemachine/ml/utilities.py

```diff
@@ -262,52 +262,51 @@ def visualize(
             model,
             feature_column_names,
             label_names,
-            tree_name,
+            size=None,
+            horizontal=False,
+            tree_name='tree',
             visual=False,
     ):
         """
         Visualize a decision tree, either in a code like format, or graphviz
         :param model: the fitted decision tree classifier
-        :param feature_column_names: column names for features
-        :param label_names: labels vector (below avg, above avg)
+        :param feature_column_names: (List[str]) column names for features
+               You can access these feature names by using your VectorAssembler (in PySpark) and calling it's .getInputCols() function
+        :param label_names: (List[str]) labels vector (below avg, above avg)
+        :param horizontal: (Bool) if the tree should be rendered horizontally
         :param tree_name: the name you would like to call the tree
         :param visual: bool, true if you want a graphviz pdf containing your file
-        :return: none
+        :return dot: The graphvis object
         """
 
         tree_to_json = DecisionTreeVisualizer.replacer(model.toDebugString,
                                                        ['feature ' + str(i) for i in
-                                                        range(0, len(feature_column_names))],
+                                                        range(len(feature_column_names)-1,-1,-1)],
```

Comment:
`range(len(feature_column_names)-1,-1,-1)`
What am I looking at with this `range` in the context of `range(start, stop, step)`?

Reply:
Yup It's instead of going from 0 -> number of columns,  it's going from number of columns -> 0 

We need to go in reverse order so we don't step on column names (ie feature1 doesn't replace feature11)

Reply:
Don't you also need to reverse the feature_column_names order? otherwise you are replacing "feature 22" with feature name 0.

Reply:
good catch, fixed